### ISSUE TITLE
Error On Update -> Unhandled Promise Rejection

### ISSUE THREAD

**Issue by salkin-mada** [buttercup/buttercup-desktop#868]
Updating from buttercup 1.16.2 to 1.18.1 on windows 10
Software Update "buttercup pop up" stuck. The "download and install" button spinning endlessly. 
```
Unhandled Promise Rejection
Error: New version 1.18.1 is not signed by the application owner: publisherNames: COMODO RSA Code Signing CA, raw info: {
  "SignerCertificate": {
    "FriendlyName": "",
    "IssuerName": {
      "Name": "CN=COMODO RSA Code Signing CA, O=COMODO CA Limited, L=Salford, S=Greater Manchester, C=GB",
      "Oid": "System.Security.Cryptography.Oid"
    },
    "NotAfter": "/Date(1578009599000)/",
    "NotBefore": "/Date(1546387200000)/",
    "PrivateKey": null,
    "PublicKey": {
      "Key": "System.Security.Cryptography.RSACryptoServiceProvider",
      "Oid": "System.Security.Cryptography.Oid",
      "EncodedKeyValue": "System.Security.Cryptography.AsnEncodedData",
      "EncodedParameters": "System.Security.Cryptography.AsnEncodedData"
    },
    "SerialNumber": "00BDC2D8815A5B41D6045C6C719D8A8F2A",
    "SignatureAlgorithm": {
      "Value": "1.2.840.113549.1.1.11",
      "FriendlyName": "sha256RSA"
    },
    "Thumbprint": "A9C56AD137B3A5359A375A1FA4827DDD071A845A",
    "Version": 3,
    "Issuer": "CN=COMODO RSA Code Signing CA, O=COMODO CA Limited, L=Salford, S=Greater Manchester, C=GB",
    "Subject": "CN=MadDev Oy, O=MadDev Oy, STREET=N�kinkaari 4 A, L=Espoo, S=Uusimaa, PostalCode=02320, C=FI"
  },
  "TimeStamperCertificate": {
    "Archived": false,
    "Extensions": [
      "System.Security.Cryptography.X509Certificates.X509KeyUsageExtension",
      "System.Security.Cryptography.X509Certificates.X509BasicConstraintsExtension",
      "System.Security.Cryptography.X509Certificates.X509EnhancedKeyUsageExtension",
      "System.Security.Cryptography.X509Certificates.X509Extension",
      "System.Security.Cryptography.X509Certificates.X509Extension",
      "System.Security.Cryptography.X509Certificates.X509SubjectKeyIdentifierExtension",
      "System.Security.Cryptography.X509Certificates.X509Extension",
      "System.Security.Cryptography.X509Certificates.X509Extension"
    ],
    "FriendlyName": "",
    "IssuerName": {
      "Name": "CN=DigiCert Assured ID CA-1, OU=www.digicert.com, O=DigiCert Inc, C=US",
      "Oid": "System.Security.Cryptography.Oid"
    },
    "NotAfter": "/Date(1729555200000)/",
    "NotBefore": "/Date(1413936000000)/",
    "HasPrivateKey": false,
    "PrivateKey": null,
    "PublicKey": {
      "Key": "System.Security.Cryptography.RSACryptoServiceProvider",
      "Oid": "System.Security.Cryptography.Oid",
      "EncodedKeyValue": "System.Security.Cryptography.AsnEncodedData",
      "EncodedParameters": "System.Security.Cryptography.AsnEncodedData"
    },
    "SerialNumber": "03019A023AFF58B16BD6D5EAE617F066",
    "SubjectName": {
      "Name": "CN=DigiCert Timestamp Responder, O=DigiCert, C=US",
      "Oid": "System.Security.Cryptography.Oid"
    },
    "SignatureAlgorithm": {
      "Value": "1.2.840.113549.1.1.5",
      "FriendlyName": "sha1RSA"
    },
    "Thumbprint": "614D271D9102E30169822487FDE5DE00A352B01D",
    "Version": 3,
    "Handle": 2804268488048,
    "Issuer": "CN=DigiCert Assured ID CA-1, OU=www.digicert.com, O=DigiCert Inc, C=US",
    "Subject": "CN=DigiCert Timestamp Responder, O=DigiCert, C=US"
  },
  "Status": 0,
  "StatusMessage": "Signature verified."
}
    at t.newError (C:/Program Files/Buttercup/resources/app.asar/dist/app.js:1:8496)
    at Object.<anonymous> (C:/Program Files/Buttercup/resources/app.asar/dist/app.js:207:228505)
    at Generator.next (<anonymous>)
    at Generator.c (C:/Program Files/Buttercup/resources/app.asar/dist/app.js:1:1411)
    at d._promiseFulfilled (C:/Program Files/Buttercup/resources/app.asar/dist/app.js:206:189112)
    at j._settlePromise (C:/Program Files/Buttercup/resources/app.asar/dist/app.js:206:145545)
    at j._settlePromise0 (C:/Program Files/Buttercup/resources/app.asar/dist/app.js:206:146143)
    at j._settlePromises (C:/Program Files/Buttercup/resources/app.asar/dist/app.js:206:147495)
    at d (C:/Program Files/Buttercup/resources/app.asar/dist/app.js:206:149218)
    at p (C:/Program Files/Buttercup/resources/app.asar/dist/app.js:206:149157)
    at s._drainQueues (C:/Program Files/Buttercup/resources/app.asar/dist/app.js:206:150677)
    at Immediate.drainQueues [as _onImmediate] (C:/Program Files/Buttercup/resources/app.asar/dist/app.js:206:148899)
From previous event:
    at j.z [as _captureStackTrace] (C:/Program Files/Buttercup/resources/app.asar/dist/app.js:206:164744)
    at new d (C:/Program Files/Buttercup/resources/app.asar/dist/app.js:206:187987)
    at C:/Program Files/Buttercup/resources/app.asar/dist/app.js:206:191076
    at t.NsisUpdater.downloadUpdate (C:/Program Files/Buttercup/resources/app.asar/dist/app.js:71:20764)
    at EventEmitter.i.ipcMain.on (C:/Program Files/Buttercup/resources/app.asar/dist/app.js:414:129520)
    at WebContents.<anonymous> (C:/Program Files/Buttercup/resources/electron.asar/browser/api/web-contents.js:285:13)
```

**Comment by salkin-mada** [comment]
and here is the log until the point posted above.

```
[2019-12-16 06:53:27.984] [info] Buttercup starting up...
[2019-12-16 06:53:28.049] [info] Restoring state... { settingsByArchiveId: {},
  currentArchive: '970eb5d1-ec69-4225-a399-a3471f12006c',
  archives:
   [ { name: 'MangoButterCup.bcup',
       id: '970eb5d1-ec69-4225-a399-a3471f12006c',
       status: 'unlocked',
       type: 'ipc',
       colour: '#000000',
       order: 0 } ],
  settings:
   { columnSizes: { tree: 230, entries: 464 },
     condencedSidebar: true,
     menubarAutoHide: false,
     archivesLoading: false,
     isTrayIconEnabled: true,
     isBrowserAccessEnabled: false,
     locale: 'en' } }
[2019-12-16 06:53:33.575] [info] Checking for update
[2019-12-16 06:53:40.080] [info] Found version 1.18.1 (url: Buttercup-Setup-1.18.1.exe)
[2019-12-16 06:53:47.411] [info] Downloading update from Buttercup-Setup-1.18.1.exe
[2019-12-16 06:53:47.416] [info] No cached update info available
[2019-12-16 06:53:47.418] [info] Download block maps (old: "https://github.com/buttercup/buttercup-desktop/releases/download/v1.16.2/Buttercup-Setup-1.18.1.exe.blockmap", new: https://github.com/buttercup/buttercup-desktop/releases/download/v1.18.1/Buttercup-Setup-1.18.1.exe.blockmap)
[2019-12-16 06:53:50.978] [error] Cannot download differentially, fallback to full download: HttpError: 404 Not Found
"method: undefined url: https://github.com/buttercup/buttercup-desktop/releases/download/v1.16.2/Buttercup-Setup-1.18.1.exe.blockmap\n\nPlease double check that your authentication token is correct. Due to security reasons actual status maybe not reported, but 404.\n"
Headers: {
  "cache-control": [
    "no-cache"
  ],
  "content-encoding": [
    "gzip"
  ],
  "content-security-policy": [
    "default-src 'none'; base-uri 'self'; connect-src 'self'; form-action 'self'; img-src 'self' data:; script-src 'self'; style-src 'unsafe-inline'"
  ],
  "content-type": [
    "text/plain; charset=utf-8"
  ],
  "date": [
    "Mon, 16 Dec 2019 05:53:50 GMT"
  ],
  "expect-ct": [
    "max-age=2592000, report-uri=\"https://api.github.com/_private/browser/errors\""
  ],
  "server": [
    "GitHub.com"
  ],
  "set-cookie": [
    "has_recent_activity=1; path=/; expires=Mon, 16 Dec 2019 06:53:50 -0000",
    "logged_in=no; domain=.github.com; path=/; expires=Fri, 16 Dec 2039 05:53:50 -0000; secure; HttpOnly",
    "_gh_sess=V1IrVjhOUHRZaWJZeUNqWFBlSTAxTkk0dzlqc0pKOHdSRzdINUV2UVhDZk0weGpzQ1dQLzNSNVlad2ZGQjBTWGZ1cWNiVTJtU1NyQmh3UjVObVprY25Ld0RJN2Vocjk3NmVCcTlvTFlYZG9UM0J4Q0JWcGJQZ2ZnMG11dkpNckh2WkpLeTByVEhNQ1p5eHUvY0l1UDJKckJjMnJ0RDBjKzdHYVc3S2NiVUFnPS0tdERnWWJqaWFxS2lpTk5UZzFPdnFTQT09--4a80e95ee246dab8446b442ec4683909ad3a74d0; path=/; secure; HttpOnly"
  ],
  "status": [
    "404 Not Found"
  ],
  "strict-transport-security": [
    "max-age=31536000; includeSubdomains; preload"
  ],
  "transfer-encoding": [
    "chunked"
  ],
  "vary": [
    "X-PJAX"
  ],
  "x-content-type-options": [
    "nosniff"
  ],
  "x-frame-options": [
    "deny"
  ],
  "x-github-request-id": [
    "C9B6:11CF6:19922FEC:2679FF5D:5DF71BEB"
  ],
  "x-request-id": [
    "76d672fa-9fc1-45df-9720-c168485b7598"
  ],
  "x-xss-protection": [
    "1; mode=block"
  ]
}
    at f (C:\Program Files\Buttercup\resources\app.asar\dist\app.js:96:7070)
    at t.ElectronHttpExecutor.handleResponse (C:\Program Files\Buttercup\resources\app.asar\dist\app.js:96:8665)
    at ClientRequest.s.doRequest.o (C:\Program Files\Buttercup\resources\app.asar\dist\app.js:96:8134)
    at ClientRequest.emit (events.js:182:13)
    at URLRequest.ClientRequest.urlRequest.on (C:\Program Files\Buttercup\resources\electron.asar\browser\api\net.js:207:12)
    at URLRequest.emit (events.js:182:13)
From previous event:
    at j.z [as _captureStackTrace] (C:\Program Files\Buttercup\resources\app.asar\dist\app.js:206:164744)
    at new d (C:\Program Files\Buttercup\resources\app.asar\dist\app.js:206:187987)
    at C:\Program Files\Buttercup\resources\app.asar\dist\app.js:206:191076
    at t.NsisUpdater.downloadUpdate (C:\Program Files\Buttercup\resources\app.asar\dist\app.js:71:20764)
    at EventEmitter.i.ipcMain.on (C:\Program Files\Buttercup\resources\app.asar\dist\app.js:414:129520)
    at EventEmitter.emit (events.js:182:13)
    at WebContents.<anonymous> (C:\Program Files\Buttercup\resources\electron.asar\browser\api\web-contents.js:285:13)
    at WebContents.emit (events.js:182:13)
[2019-12-16 06:54:06.548] [warn] Sign verification failed, installer signed with incorrect certificate
```

**Comment by salkin-mada** [comment]
fixed by uninstalling 1.16.2 and installing 1.18.x with the new 'SignerCertificate'
